storage/cache/log_subtree_cache.go

```diff
@@ -59,42 +102,31 @@ func populateLogSubtreeNodes(hasher hashers.LogHasher) storage.PopulateSubtreeFu
 			st.InternalNodes = make(map[string][]byte)
 		}
 
-		// We need to update the subtree root hash regardless of whether it's fully populated
-		for leafIndex := int64(0); leafIndex < int64(len(st.Leaves)); leafIndex++ {
-			nodeID := storage.NewNodeIDFromPrefix(st.Prefix, logStrataDepth, leafIndex, logStrataDepth, maxLogDepth)
-			_, sfx := nodeID.Split(len(st.Prefix), int(st.Depth))
-			sfxKey := sfx.String()
-			h := st.Leaves[sfxKey]
-			if h == nil {
-				return fmt.Errorf("unexpectedly got nil for subtree leaf suffix %s", sfx)
+		// Special case the perfect subtree case for now:
+		if len(st.Leaves) == maxLeaves {
+			if err := populatePerfectLogSubtree(hasher, st); err != nil {
+				return err
 			}
-			seq, err := cmt.AddLeafHash(h, func(height int, index int64, h []byte) error {
-				if height == logStrataDepth && index == 0 {
-					// no space for the root in the node cache
-					return nil
+		} else {
+			// We need to update the subtree root hash regardless of whether it's fully populated
+			for leafIndex := int64(0); leafIndex < int64(len(st.Leaves)); leafIndex++ {
+				nodeID := storage.NewNodeIDFromPrefix(st.Prefix, logStrataDepth, leafIndex, logStrataDepth, maxLogDepth)
```

Comment:
We should be able to avoid that call to `NewNodeIDFromPrefix` as well? I think it only uses the Suffix.